packages/cra-external-template/template/src/core/connect-session.tsx

```diff
@@ -0,0 +1,12 @@
+import { ReapitConnectBrowserSession } from '@reapit/connect-session'
+import { Config } from '../types/global'
+
+export const ReapitConnectBrowserSessionInstance = {
```

Comment:
I like this implementation but why not just do something like:

```
const ReapitConnectBrowserSessionInstance = {
  reapitConnectSession: ReapitConnectBrowserSession,
  initReapitConnectSession: function(config: Config) {
    this.instance = new ReapitConnectBrowserSession({
      connectClientId: config.cognitoClientId,
      connectOAuthUrl: config.cognitoOAuthUrl,
    })
  },
}

export const { reapitConnectSession, initReapitConnectSession } = ReapitConnectBrowserSessionInstance

```

Then in your code you can just import the session or the initiailizer

Reply:
This wont work as expected. 

`reapitConnectSession` that we export as above will be just a reference. When we call `initReapitConnectSession`. The real `reapitConnectSession` will be assigned a different value, and the reference is still point to the old value: null

Reply:
I would love to keep this implementation, but replace `reapitConnectSession` with getter version incase we need to mock it differently for difference test cases!?

Reply:
E.g. `jest.spyOn(ReapitConnectBrowserSessionInstance, 'ReapitConnectBrowserSessionInstance') and `mockImplementation` combo?

Reply:
> This wont work as expected.
> 
> `reapitConnectSession` that we export as above will be just a reference. When we call `initReapitConnectSession`. The real `reapitConnectSession` will be assigned a different value, and the reference is still point to the old value:

I would need to check this but you can definitely `export default ReapitConnectBrowserSessionInstance. reapitConnectSession` which would have a similar effect of tidying upt thr code

Reply:
> E.g. `jest.spyOn(ReapitConnectBrowserSessionInstance, 'ReapitConnectBrowserSessionInstance')` and `mockImplementation` combo?

Just put a file in the __mocks__ folder, and use jest.mock

Or in the above example you can `jest.spyOn(reapitConnectSession, 'default')` if you prefer